### PR TITLE
Add Install page

### PR DESCRIPTION
**Pull request by straight-shoota**
Implements a new `Install` page overview and updates the install subpages, drastically cutting down on duplicate content there.

The main page has simple instructions directly available for a ton of installation mechanisms, as well as direct links to further instructions and package information.
The page is pretty plain and simple (no interactive selection as we had previously toyed with), but it's very usable and rich on information. A clear structure based on the operating system and then categorized by the type of package provider (official Crystal packages, system packages, 3rd party community packages) enables finding useful solutions easily.

The sub pages are trimmed down and each one has now exactly one purpose, detailing the installation instructions for the official packages on a specific operating system, or instructions for installation with a specific package manager. There are no longer individual pages for specific linux distribution which all shared more or less the same code. They redirect to the linux overview now, which should even be more helpful.

Further enhancements of this page could introduce some kind of platform-detection mechanism to highlight the current operating system, for example to pin instructions for the current OS to the top of the page.

<del>Depends on #529 and #534</del>

grafik

### CODE REVIEW

**Comment by beta-ziliani** [comment]
Love the concept so far! 🤩

**Comment by straight-shoota** [comment]
Screenshot after some design iterations with @jkicillof:

grafik

**Comment by straight-shoota** [comment]
I have incorporated the relevant visual changes from #547 into this branch, which is mostly responsive layout.
I kept the plain (i.e. not button) styling for the titles, and did not port any of the internal refactorings.

**Comment by straight-shoota** [comment]
I have rebased on the relaunch branch without any dependency PRs. This should be ready to merge now.

grafik

**Review by beta-ziliani — CHANGES_REQUESTED** [review]
Overall I love it. I have several improvements to require (or suggest, happy to discuss them), in order of importance:
 - Shouldn't "Build from Source" and "Nightly" be in the first section (Crystal)? The tarball _is_ building from source, so it's weird to have it twice and disconnected. Alternatively, we can put both, the "tarball" and "nightly" in each relevant section (Linux, macOS, Window (which is missing)) pointing at the corresponding #section in the page.
 - The info box for 84codes' `packagecloud.io` should go in the text, as it is required information and not just additional info (as in the others).
 - Same with the internal links to specific installation modes, I think we need to link them within the text those that _require_ extra steps (instead of listing the steps), or place them in the info box if this is just additional information (like the scoop package). Then, the info about the package should be (as it is now) in the specific page instead of in the info box.
 - the color in the bash code makes no sense, better plain text. The scoop one also looks weird.
 - Can we add a green `latest` for those we know are updated? (snapcraft, installers, etc, which we control).

**Comment by beta-ziliani** [comment]
One more: I like how Julia adds the tier next to the platform, maybe we can consider adding something like that

**Comment by straight-shoota** [comment]
* IMO nightlies and source builds are entirely different categories. I could see nightlies being understood as just a different release channel than stable. That's usually how the nightly packages work. It probably still makes sense to have a separate overview of which package sources provide nightlies because not all of them do.
* Agreed
* Internal links will be better noticable once the updated link styles are merged.
* Updating the color scheme for the highlighter is on the TODO list. This is an independent issue.
* I'll add a version indicator for the official packages.

File: _pages/install.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    <h3>System</h3>
+    <div class="install-entries">
+      <div class="install-entry">
+        <span class="title"><a href="on_openbsd/#install-package">Package</a></span>
+        {% highlight shell %}doas pkg_add crystal{% endhighlight %}
+      </div>
+      <div class="install-entry">
+        <span class="title"><a href="on_openbsd/#install-port">Port</a></span>
+        {% highlight shell %}doas make -C/usr/ports/lang/crystal clean install{% endhighlight %}
+        <a href="https://openports.pl/path/lang/crystal" title="Crysta port on openports.pl" class="info">{% include icons/info.svg %}</a>
```
Comment (beta-ziliani) [inline _pages/install.md:223]:
```suggestion
        <a href="https://openports.pl/path/lang/crystal" title="Crystal port on openports.pl" class="info">{% include icons/info.svg %}</a>
```

File: _pages/install.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    <div class="install-entries">
+      <div class="install-entry">
+        <span class="title"><a href="on_freebsd/#install-package">Package</a></span>
+        {% highlight shell %}sudo pkg install -y crystal shards{% endhighlight %}
+      </div>
+      <div class="install-entry">
+        <span class="title"><a href="on_freebsd/#install-port">Port</a></span>
+        {% highlight shell %}sudo make -C/usr/ports/lang/crystal reinstall clean
+sudo make -C/usr/ports/devel/shards reinstall clean
+        {% endhighlight %}
```
Comment (beta-ziliani) [inline _pages/install.md:202]:
```suggestion
sudo make -C/usr/ports/devel/shards reinstall clean{% endhighlight %}
```

**Review by beta-ziliani — CHANGES_REQUESTED** [review]
Some minor fixes left.

I still think we can improve the consistency of the three data cells (name, instructions, info icon), when to use which and for what. And add the tier. But I will leave that for a second iteration.

File: assets/install/version-badge.svg [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+layout: none
+---
+{% assign latest_release = site.releases | reverse | first %}
+<svg xmlns="http://www.w3.org/2000/svg" width="46" height="20">
+<clipPath id="clip"><rect rx="3" width="100%" height="100%" fill="#000"/></clipPath>
+<linearGradient id="grad" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><g clip-path="url(#clip)"><rect width="100%" height="100%" fill="#555"/><rect x="0" y="0" width="46" height="20" fill="#4c1"/><rect y="0" width="100%" height="20" fill="url(#grad)"/><g fill="#fff" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11"><text x="23.0" y="15" fill="#010101" fill-opacity=".3" text-anchor="middle">{{ latest_release.version }}</text><text x="23.0" y="14" text-anchor="middle">{{ latest_release.version }}</text></g></g>
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply assets/install/version-badge.svg:7]: I think GitHub primarily complains about the YAML frontmatter 😆
I'll change the file extension.

[the remark this answers is not in the document]
Comment (beta-ziliani) [reply assets/install/version-badge.svg]: ahh! that red is not a diff! that's confusing! (although I should've noted it)

**Review by beta-ziliani — APPROVED** [review]
🚀

**Pull request merged**